### ISSUE TITLE
Kickout when locking or logging off (even on)

### ISSUE THREAD

**Issue by rossmarkflo** [Ylianst/MeshCentral#3090]
When logging off or locking a device we get kicked out - not sure if this is due to the consent but can this be bypassed?

Saw same happing on non-consent device when logging on, always get kicked off and have to reconnect.

There should be at-least a auto-reconnect if it can't be avoided - if that's a possible work-around..

**Comment by PathfinderNetworks** [comment]
This is almost certainly the same root issue as reported in #3028

**Comment by krayon007** [comment]
It shouldn't happen when you lock the screen, I can see if I can reproduce that. Do you have RDP sessions connected when you lock? I know RDP is a special case...

As far as logging off, the main contention is that the KVM is running as the logged in user. So when you logout, the screen capture fails (or is supposed to fail), but the process lacks permission to respawn as root, which is required to capture the login screen. I've seen a few scenarios where after logoff, the screen capture doesn't fail, which is why the parent process can't detect that something happened if that happens. One thing I can do, is modify the parent to watch for login/logoff events, and try to auto reconnect based on that instead of only relying on a security error. I'll setup a test scenario for that

**Comment by Ylianst** [comment]
With MeshCentral now supporting locking the desktop on disconnect and the upcoming version 0.9.18 that has the context menu on the disconnect button, I have been generating a lot more "lock" commands and I have gotten the agent to crash. The "MeshAgent.log" on Windows 10, 64bit agent reports this:

```
[2021-09-01 12:16:33 PM] [80645B8D669A2A74] ..\microstack\ILibParsers.c:2711 (0,0) FATAL EXCEPTION @ [FuncAddr: 0x00007ff605692dae / BaseAddr: 0x00007ff6056f791c / Delta: 412526]
```

Can other report that the "kickout" is actually the agent crashing? If so, once the agent reconnects, can you download the "meshagent.log" file and report back if any errors occurred? - Thanks.

**Comment by krayon007** [comment]
Ok, so the main reason the KVM is kicking you out when you log-out, is if you have the connection-bar enabled. The intended behavior is if the connection bar is closed, it will force the KVM session to close. When you logout, the connection bar process is force closed by the OS, which is what is closing the KVM session. This was sort of intentional, because we wanted to make it so if someone simply killed the connection bar process, it would close the KVM session. So in this particular case, the OS is killing the process on logout...

With that being said, I was able to reproduce a scenario where when I disabled the connection bar, I get the yellow message saying the KVM child process has unexpectedly closed... I think I can fix that, and have it re-launch the child to the GDM session. I have to debug why it isn't doing that now, becuase the code that respawns the child into the user process on login, should already handle this particular case. So I'm looking into that now.

As far as the lock screen kicking you out. What distro are you using? I ask because I forget which distro (I have 72 test systems), but I remember a few days ago I noticed one of them the lock-screen was running as a different X session as the user desktop session, which caused the child process to exit when it lacked permission to scrape the lockscreen. I'll look into this case after I fix the above scenario I mentioned earlier.

**Comment by krayon007** [comment]
As a side note, the issue with the child process exiting on login, should be resolved in server version v0.9.22. That issue was caused by a resolution change causing an X error when I scraped the session. I found that on some of my systems, I had the desktop resolution set to a resolution that was different than the login screen resolution. This made it so when you logged in, as soon as the workspace activated and changed the resolution, the X client exited because of a configuration mismatch. I found that on some distros this config mismatch caused X to exit, but on other distros, it simply caused X to just scale the screen capture... But regardless, this issue was fixed in v0.9.22
